modules/spark/src/main/scala/org/apache/ignite/spark/impl/package.scala

```diff
@@ -88,42 +89,48 @@ package object impl {
     /**
       * @param ignite Ignite instance.
       * @param tabName Table name.
+      * @param schemaName Optional schema name.
       * @return True if table exists false otherwise.
       */
-    def sqlTableExists(ignite: Ignite, tabName: String): Boolean =
-        sqlTableInfo(ignite, tabName).isDefined
+    def sqlTableExists(ignite: Ignite, tabName: String, schemaName: Option[String]): Boolean =
+        sqlTableInfo(ignite, tabName, schemaName).isDefined
 
     /**
       * @param ignite Ignite instance.
       * @param tabName Table name.
+      * @param schemaName Optional schema name.
       * @return QueryEntity for a given table.
       */
-    def igniteSQLTable(ignite: Ignite, tabName: String): Option[QueryEntity] =
-        sqlTableInfo[Any, Any](ignite, tabName).map(_._2)
+    def igniteSQLTable(ignite: Ignite, tabName: String, schemaName: Option[String]): Option[QueryEntity] =
+        sqlTableInfo[Any, Any](ignite, tabName, schemaName).map(_._2)
 
     /**
       * @param ignite Ignite instance.
       * @param tabName Table name.
+      * @param schemaName Optional schema name.
       * @return Cache name for given table.
       */
-    def sqlCacheName(ignite: Ignite, tabName: String): Option[String] =
-        sqlTableInfo[Any, Any](ignite, tabName).map(_._1.getName)
+    def sqlCacheName(ignite: Ignite, tabName: String, schemaName: Option[String]): Option[String] =
+        sqlTableInfo[Any, Any](ignite, tabName, schemaName).map(_._1.getName)
 
     /**
       * @param ignite Ignite instance.
       * @param tabName Table name.
+      * @param schemaName Optional schema name.
       * @tparam K Key class.
       * @tparam V Value class.
       * @return CacheConfiguration and QueryEntity for a given table.
       */
-    def sqlTableInfo[K, V](ignite: Ignite, tabName: String): Option[(CacheConfiguration[K, V], QueryEntity)] =
-        ignite.cacheNames().map { cacheName ⇒
-            val ccfg = ignite.cache[K, V](cacheName).getConfiguration(classOf[CacheConfiguration[K, V]])
-
-            val queryEntities = ccfg.getQueryEntities
-
-            queryEntities.find(_.getTableName.equalsIgnoreCase(tabName)).map(qe ⇒ (ccfg, qe))
-        }.find(_.isDefined).flatten
+    def sqlTableInfo[K, V](ignite: Ignite, tabName: String,
+        schemaName: Option[String]): Option[(CacheConfiguration[K, V], QueryEntity)] =
+        ignite.cacheNames()
+            .map(cacheName => ignite.cache[K, V](cacheName).getConfiguration(classOf[CacheConfiguration[K, V]]))
+            .filter(
+                ccfg => schemaName.isEmpty
```

Comment:
can be simplified to

`schemName.forall(_.equalsIgnoreCase(normalizeSchemaName(ccfg.getName, ccfg.getSqlSchema))))`